src/common/cmdline-utils.cpp

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2020 [Ribose Inc](https://www.ribose.com).
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+ *
+ * 1.  Redistributions of source code must retain the above copyright notice,
+ *     this list of conditions and the following disclaimer.
+ *
+ * 2.  Redistributions in binary form must reproduce the above copyright notice,
+ *     this list of conditions and the following disclaimer in the documentation
+ *     and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+/** Command Line utilities
+ *  @file
+ */
+
+#ifdef _WIN32
+#include "cmdline-utils.h"
+#include <windows.h>
+#include <locale>
+#include <codecvt>
+#include <vector>
+
+static bool
+contains_non_ascii(int argc, char **argv)
+{
+    for (int i = 0; i < argc; i++) {
+        for (auto sz = argv[i]; *sz; sz++) {
+            if ((*sz & 0x7F) != *sz) {
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+static std::vector<std::string>
+get_utf8_args()
+{
+    int       arg_nb;
+    wchar_t **arg_w;
+
+    arg_w = CommandLineToArgvW(GetCommandLineW(), &arg_nb);
+    if (!arg_w) {
+        throw std::runtime_error("CommandLineToArgvW failed");
+    }
+
+    std::vector<std::string> result;
+    result.reserve(arg_nb);
+
+    std::wstring_convert<std::codecvt_utf8<wchar_t>> utf8conv;
+    for (int i = 0; i < arg_nb; i++) {
+        auto utf8 = utf8conv.to_bytes(arg_w[i]);
+        result.push_back(utf8);
+    }
```

Comment:
Shouldn't we wrap this in try ... catch so `LocalFree()` is called even when exception is raised?

Reply:
Done